src/components/date-picker/flatpickr-adapter/flatpickr-adapter.tsx

```diff
@@ -0,0 +1,188 @@
+
+@Component({
+    tag: 'limel-flatpickr-adapter',
+    shadow: true,
+    styleUrl: 'flatpickr-adapter.scss',
+})
+export class DatePickerCalendar {
+    /**
+     * The value of the field.
+     */
+    @Prop()
+    public value: Date;
+
+    /**
+     * Type of date picker. Defaults to `datetime`
+     */
+    @Prop()
+    public type: DateType = 'datetime';
+
+    /**
+     * Format to display the selected date in
+     */
+    @Prop()
+    public format: string;
+
+    @Prop()
+    public isOpen: boolean;
+
+    @Prop()
+    public inputElement: HTMLElement;
+
+    /**
+     * Defines the localisation for translations and date formatting.
+     * Property `format` customizes the localized date format.
+     */
+    @Prop()
+    public language: Languages = 'en';
+
+    /**
+     * Emitted when the date picker value is changed.
+     */
+    @Event()
+    public change: EventEmitter<Date>;
+
+    private picker: Picker;
+    private flatPickrCreated: boolean = false;
+
+    private container: HTMLElement;
+
+    public componentWillLoad() {
+        switch (this.type) {
+            case 'date':
+                this.picker = new DateOnlyPicker(
+                    this.format,
+                    this.language,
+                    this.change
+                );
+                break;
+
+            case 'time':
+                this.picker = new TimePicker(
+                    this.format,
+                    this.language,
+                    this.change
+                );
+                break;
+
+            case 'week':
+                this.picker = new WeekPicker(
+                    this.format,
+                    this.language,
+                    this.change
+                );
+                break;
+
+            case 'month':
+                this.picker = new MonthPicker(
+                    this.format,
+                    this.language,
+                    this.change,
+                    translate
+                );
+                break;
+
+            case 'quarter':
+                this.picker = new QuarterPicker(
+                    this.format,
+                    this.language,
+                    this.change,
+                    translate
+                );
+                break;
+            case 'year':
+                this.picker = new YearPicker(
+                    this.format,
+                    this.language,
+                    this.change,
+                    translate
+                );
+                break;
+
+            case 'datetime':
+            default:
+                this.picker = new DatetimePicker(
+                    this.format,
+                    this.language,
+                    this.change
+                );
+                break;
+        }
+    }
+
+    public componentDidUpdate() {
+        if (this.flatPickrCreated) {
+            this.redrawFlatpickr();
+        } else {
+            this.createFlatpickr();
+        }
```

Comment:
I think we might not need this anymore, but I'm leaning towards leaving it until we've merged, and then look at whether it's possible to remove afterwards 🙂